docs/fundamentals/networking/networking-telemetry.md

```diff
@@ -0,0 +1,547 @@
+- `Microsoft.AspNetCore.Hosting`
+- `Microsoft-AspNetCore-Server-Kestrel`
+
+The telemetry has some performance overhead when enabled, so make sure to subscribe only to groups you are actually interested in.
+
+## Event Counters (metrics)
+
+[EventCounters] are .NET APIs used for lightweight, cross-platform, and near real-time performance metric collection.
+
+Networking components are instrumented to publish basic diagnostics information using EventCounters.
+They include information like
+
+- `System.Net.Http` > `requests-started`
+- `System.Net.Http` > `requests-failed`
+- `System.Net.Http` > `http11-connections-current-total`
+- `System.Net.Security` > `all-tls-sessions-open`
+- `System.Net.Sockets` > `outgoing-connections-established`
+- `System.Net.NameResolution` > `dns-lookups-duration`
+
+**For the full list, check out [well-known counters].**
+
+### Monitor metrics from outside the process
+
+#### dotnet-counters
+
+[`dotnet-counters`][dotnet-counter docs] is a cross-platform performance monitoring tool for ad-hoc health monitoring and first-level performance investigation.
+
+```console
+dotnet tool install --global dotnet-counters
+```
+
+```console
+dotnet-counters monitor --counters System.Net.Http,System.Net.Security --process-id 1234
+```
+
+The command will keep refreshing the console with the latest numbers
+
+```txt
+[System.Net.Http]
+    Current Http 1.1 Connections                       3
+    Current Http 2.0 Connections                       1
+    Current Http 3.0 Connections                       0
+    Current Requests                                   4
+    HTTP 1.1 Requests Queue Duration (ms)              0
+    HTTP 2.0 Requests Queue Duration (ms)              0
+    HTTP 3.0 Requests Queue Duration (ms)              0
+    Requests Failed                                    0
+    Requests Failed Rate (Count / 1 sec)               0
+    Requests Started                                 470
+    Requests Started Rate (Count / 1 sec)             18
+```
+
+See [dotnet-counter docs] for all the available commands and parameters.
+
+#### Application Insights
+
+Application Insights does not collect event counters by default.
+See [AppInsights EventCounters docs] on customizing the set of counters you are interested in.
+
+For example:
+
+```c#
+services.ConfigureTelemetryModule<EventCounterCollectionModule>((module, options) =>
+{
+    module.Counters.Add(new EventCounterCollectionRequest("System.Net.Http", "current-requests"));
+    module.Counters.Add(new EventCounterCollectionRequest("System.Net.Http", "requests-failed"));
+    module.Counters.Add(new EventCounterCollectionRequest("System.Net.Http", "http11-connections-current-total"));
+    module.Counters.Add(new EventCounterCollectionRequest("System.Net.Security", "all-tls-sessions-open"));
+});
+```
+
+See [this sample](https://gist.github.com/MihaZupan/02ec402fa34880df129351b92e18e86c) for subscribing to many runtime and ASP.NET event counters.
```

Comment:
Does it need to be a gist? Can we just add it as a code file to this repo instead, and reference it as a code snippet?